### PR TITLE
Adding variance to scil_visualize_fodf

### PR DESCRIPTION
**Pull request by karanphil**
This is simply an attempt to merge an old branch of @CHrlS98  into the main script `scil_visualize_fodf.py`. With my simulations, I have a lot of repetitions of the same fODF reconstruction so I compute the variance of it. This addition allows us to view the variance as another fODF behind the "real" one, with the color of your choice. Note that you need to input the mean + some factor of the variance to get the right visual effect. 

Here is some data to test it. Also, it would be useful if users of this script try a typical command to confirm that I didn't break anything. @CHrlS98 @jhlegarreta 

https://usherbrooke-my.sharepoint.com/:f:/g/personal/karp2601_usherbrooke_ca/EuPAjEN6VHlNi4-MGW3rjwcBCLNpkCRNuMD1F_K3PRxH8w?e=wY35T7

With this data, try: `scil_visualize_fodf.py wm_fodf_mean.nii.gz --variance wm_fodf_var.nii.gz --axis_name coronal --sph_subdivide 2 --bg_color 1 1 1 --color_rgb 0 0.65 0.35 --var_color 0.6 0.6 0.6`, this will show the fODFs with a grey outline representing the fODF + 2 * std.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scilpy/viz/scene_utils.py [hunk 1]
```diff
@@ -158,6 +159,10 @@ def create_odf_slicer(sh_fodf, orientation, slice_index, mask, sphere,
         If True, enables normalization of ODF slicer.
     colormap : str
         Colormap for the ODF slicer. If None, a RGB colormap is used.
+    sh_variance : np.ndarray, optional
+        Spherical harmonics of the variance fODF data.
+    variance_color : tupple, optional
```
Comment (jhlegarreta) [inline scilpy/viz/scene_utils.py:164]: typo: tupple -> tuple

**Review by jhlegarreta — COMMENTED** [review]
Thanks for doing this @karanphil.

I seem not to have access to the shared data, so I could not test it. I'd be willing to give it a try if you put the data available again.

Otherwise, I tested the mosaic overlap script with structural data only and the changes introduced in here do not seem to have undesired side-effects.

**Comment by karanphil** [comment]
> Thanks for doing this @karanphil.
> 
> I seem not to have access to the shared data, so I could not test it. I'd be willing to give it a try if you put the data available again.
> 
> Otherwise, I tested the mosaic overlap script with structural data only and the changes introduced in here do not seem to have undesired side-effects.

It should work now!

**Review by jhlegarreta — APPROVED** [review]
> It should work now!

Have just tested. The provided data and command work well for the purpose. Nice Philippe.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by karanphil** [comment]
Works fine with python 3.10!

File: scripts/scil_visualize_fodf.py [hunk 2]
```diff
@@ -146,6 +146,13 @@ def _build_arg_parser():
                                    help='Length of the peaks segments. '
                                         '[%(default)s]')
 
+    # fODF variance options
+    p.add_argument('--variance',
+                   help='FODF variance (mean + k * variance) file.')
```
Comment (frheault) [inline scripts/scil_visualize_fodf.py:151]: Here there is some documentation needed. You should explain why k is needed (I believe it is not). 
You could explain that you can use a factor (maybe with scil_image_math.py) for visualization reason or what not.

But this k factor here is not clear.

**Review by frheault — CHANGES_REQUESTED** [review]
After my comment: Good to go

File: scripts/scil_visualize_fodf.py (hunk 2 above)
Comment (karanphil) [reply scripts/scil_visualize_fodf.py]: Is this clearer?

Comment (arnaudbore) [reply scripts/scil_visualize_fodf.py]: Put k as a variable. I'm more comfortable to ask for variance file and factor than a file containing k * variance.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_visualize_fodf.py (hunk 2 above)
Comment (karanphil) [reply scripts/scil_visualize_fodf.py]: I changed it!

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Review by arnaudbore — APPROVED** [review]

**Comment by karanphil** [comment]
> This is simply an attempt to merge an old branch of @CHrlS98 into the main script `scil_visualize_fodf.py`. With my simulations, I have a lot of repetitions of the same fODF reconstruction so I compute the variance of it. This addition allows us to view the variance as another fODF behind the "real" one, with the color of your choice. Note that you need to input the mean + some factor of the variance to get the right visual effect.
> 
> Here is some data to test it. Also, it would be useful if users of this script try a typical command to confirm that I didn't break anything. @CHrlS98 @jhlegarreta
> 
> https://usherbrooke-my.sharepoint.com/:f:/g/personal/karp2601_usherbrooke_ca/EuPAjEN6VHlNi4-MGW3rjwcBCLNpkCRNuMD1F_K3PRxH8w?e=wY35T7
> 
> With this data, try: `scil_visualize_fodf.py wm_fodf_mean.nii.gz --variance wm_fodf_mean_2std.nii.gz --axis_name coronal --sph_subdivide 2 --bg_color 1 1 1 --color_rgb 0 0.65 0.35 --var_color 0.6 0.6 0.6`, this will show the fODFs with a grey outline representing the fODF + 2 * std.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**